### PR TITLE
[v2.4] Pass Debug Flag to Machine If Trace Enabled

### PR DESCRIPTION
**Pull request by luthermonson**
Update Machine to [rancher46](https://github.com/rancher/machine/releases/tag/v0.15.0-rancher46) and update for passthru of a `--debug` flag if trace is enabled.

### CODE REVIEW

File: pkg/controllers/management/node/utils.go [hunk 1]
```diff
@@ -130,6 +136,8 @@ func buildCommand(nodeDir string, node *v3.Node, cmdArgs []string) (*exec.Cmd, e
 		"PATH=/usr/bin:/var/lib/rancher/management-state/bin",
 	}
 	command.Env = jailer.WhitelistEnvvars(envvars)
+
+	logrus.Tracef("buildCommand args: %v", command.Args)
```
Comment (dramich) [inline pkg/controllers/management/node/utils.go:140]: Is just the args enough? Previous trace log was the whole thing

Comment (luthermonson) [reply pkg/controllers/management/node/utils.go:140]: its actually more, includes `rancher-machine` on the front. it's everything needed tho

**Comment by luthermonson** [comment]
https://github.com/rancher/rancher/issues/29519

**Comment by luthermonson** [comment]
https://github.com/rancher/rancher/issues/25404
https://github.com/rancher/rancher/issues/27732

**Comment by luthermonson** [comment]
https://github.com/rancher/rancher/issues/27699

File: pkg/controllers/management/node/utils.go [hunk 2]
```diff
@@ -188,18 +196,25 @@ func getSSHKey(nodeDir, keyPath string, obj *v3.Node) (string, error) {
...
-		logrus.Debugf("stdout: %s", msg)
+
 		_, err := filterDockerMessage(msg, node)
 		if err != nil {
 			return node, err
 		}
-		logrus.Infof("[node-controller-rancher-machine] %v", msg)
-		v3.NodeConditionProvisioned.Message(node, msg)
+		if strings.HasPrefix(msg, debugPrefix) {
+			// calls in machine with log.Debug all prefixed and are spammy, leave them in trace
```
Comment (dramich) [inline pkg/controllers/management/node/utils.go:211]:
```suggestion
			// calls in machine with log.Debug are all prefixed and spammy so only log under trace and don't add to the v3.NodeConditionProvisioned.Message
```

**Review by dramich — APPROVED** [review]

**Pull request merged**